### DISCUSSION TITLE
FluidProperties module: possible bug in MethaneFluidProperties

### DISCUSSION THREAD

**Discussion by garciapintado** [idaholab/moose#23231] [Q&A Modules: General] [answered]
Hi all,
Just noted that a PorousFlow simulation using `MethaneFluidProperties` from the Fluid Properties module was having a negative specific enthalpy, and wondered if that is right. So, I've obtained some tabulated fluid properties for methane and compared those with those obtained by CoolProp [both cite Setzmann and Wagner (1991) as reference]. Both enthalpy and internal energy differ wildly from those obtained from CoolProp [right column here]: 
<img width="1623" alt="Screenshot 2023-01-23 at 20 49 00" src="https://user-images.githubusercontent.com/44291802/214135643-fc4e89aa-782d-47a6-90bf-514a719b8b0f.png">

Also viscosity [although with different references] is rather different and strangely non-dependent on pressure within FluidProperties:
<img width="1074" alt="Screenshot 2023-01-23 at 20 54 44" src="https://user-images.githubusercontent.com/44291802/214136710-3bf0df24-3bb8-4e75-baef-1f4fd84fb493.png">

I've tested another gas (hydrogen) as example, and the EOS give similar results in both codes. This is absolutely not a priority for me. Just wanted to report in case it is useful...

**Comment by GiudGiud** [comment, marked as answer]
@cpgr any insights on this comparison?

For enthalpy and energy values we could work with negative values, it's not wrong it just means the reference energy/enthalpy has been set differently. 
The viscosity is directly more concerning (but we ll still have a look at enthalpy and energy)

**Reply by cpgr** [reply]
I'll take a look. Like you say, enthalpy and internal energy could just be referenced from a different state.

**Reply by cpgr** [reply]
So, `MethaneFluidProperties` uses the same reference state for enthalpy and internal energy as Setzmann and Wagner (1991) - which is T = 298.15 K, P = 101.325 kPa - and exactly produces the tables of values provided in that reference.

In contrast, CoolProp use the same reference state as REFPROP, which is the boiling point T = 111.667 K. This explains the difference in absolute values.

It looks like I implemented a simple viscosity correlation, rather than the much more complicated correlation implemented in CoolProp. I haven't compared either to experimental values I don't think.

**Reply by garciapintado** [reply]
Ah, OK! Many thanks for the explanation!
